diagnostic_menu.py

```diff
@@ -0,0 +1,181 @@
+from datetime import date
+from PyQt5.QtCore import QTimer, Qt, pyqtSignal
+from PyQt5.QtGui import QFont
+import logging
+from PyQt5.QtWidgets import QMainWindow, QVBoxLayout, QWidget, QPushButton, QLabel, QScrollArea
+import shutil
+import psutil 
+
+ASSETS_FOLDER = "./assets"
+
+
+logger = logging.getLogger('diagnostic_diagnosticmenu')
+
+
+class DataScrollArea(QScrollArea):
+    def __init__(self, video_data):
+        super().__init__()
+        self.setWindowTitle('Logs')
+        self.setGeometry(400, 400, 700, 500)
+        self.setStyleSheet("background-color: black;") 
+
+        self.data = video_data
+
+        self.json_label = QLabel(text=f'{self.data}')
+        self.json_label.setStyleSheet('color: white; font-weight: bold;')
+
+
+        layout = QVBoxLayout()
+        layout.setAlignment(Qt.AlignTop)
+        layout.addWidget(self.json_label)
+
+        widget = QWidget()
+        widget.setLayout(layout)
+        self.setWidget(widget)
+        self.setWidgetResizable(True)
+
+    def keyPressEvent(self, qKeyEvent):
+        if qKeyEvent.key() == Qt.Key_Escape: 
+            self.close()
+        else:
+            super().keyPressEvent(qKeyEvent)
+
+class LogScrollArea(QScrollArea):
+    def __init__(self):
+        super().__init__()
+        self.setWindowTitle(f'Log file: {date.today()}.log')
+        self.setGeometry(400, 400, 700, 500)
+        self.setStyleSheet("background-color: black;") 
+
+        self.logs = 'Test'
+
+        self.logs_label = QLabel(text=f'{self.logs}')
+        self.logs_label.setStyleSheet('color: white; font-weight: bold;')
+
+        self.updateLogs(f'{ASSETS_FOLDER}/logs/{date.today()}.log')
+
+
+        layout = QVBoxLayout()
+        layout.setAlignment(Qt.AlignTop)
+        layout.addWidget(self.logs_label)
+
+        widget = QWidget()
+        widget.setLayout(layout)
+        self.timer = QTimer(self)
+
+        self.timer.timeout.connect(lambda: self.updateLogs(f'{ASSETS_FOLDER}/logs/{date.today()}.log'))
+        self.timer.start(1000)
+        
+
+        self.setWidget(widget)
+        self.setWidgetResizable(True)
+  
+    def updateLogs(self, file):
+        try:
+            with open(file, 'r') as txt:
+                list_of_lines = txt.read() 
+            self.logs = list_of_lines
+        except FileNotFoundError:
+            logger.error(f'File {file} not found')
+            self.logs = 'File not found'
+        self.logs_label.setText(list_of_lines)
+       
+    def keyPressEvent(self, qKeyEvent):
+        if qKeyEvent.key() == Qt.Key_Escape: 
+            self.close()
+        else:
+            super().keyPressEvent(qKeyEvent)
+
+class DiagnosticMenu(QMainWindow):
```

Comment:
"Diagnostic menu" is kind of a beaten up phrase, but generally a menu is a window providing options, but not containing the options themselves. This may cause a slight confusion, making the reader believe that the actual diagnostic functionality is located somewhere else (but it instead is a part of this window as tabs).

I understand that this may sound way too nitpicky, but there is a reason I care about naming as much. I'll give you a rant on it over the next call.